aspnet/mvc/views/view-components.rst

```diff
@@ -18,57 +27,91 @@ New to ASP.NET MVC 6, view components are similar to partial views, but they are
 
-Examine the ViewComponent class
---------------------------------
+A view component class:
 
-- Examine the *src\\TodoList\\ViewComponents\\PriorityListViewComponent.cs* file:
+- Fully supports constructor `Dependency Injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`__
+- Does not take part in the controller lifecycle, which means you can't use action filters in a view component
 
-  .. code-block:: c#
-    :linenos:
+View component methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A view component defines its logic in an ``InvokeAsync`` method that returns an `IViewComponentResult <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/IViewComponentResult/index.html>`__. Parameters come directly from invocation of the view component, not from model binding. A view component never directly handles a request. Typically a view component initializes a model and passes it to a view by calling the `View <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewComponent/index.html>`__ method. In summary, view component methods:
+
+- Define an `InvokeAsync`` method that returns an ``IViewComponentResult``
+- Typically initializes a model and passes it to a view by calling the `ViewComponent <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewComponent/index.html>`__  `View <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ViewResult/index.html>`__ method
+- Parameters come from the calling method, not HTTP, there is no model binding
+- Are not reachable directly as an HTTP endpoint, they are invoked from your code (usually in a view). A view component never handles a request
+- Are overloaded on the signature rather than any details from the current HTTP request
+
+View search path
+^^^^^^^^^^^^^^^^^^
+
+The runtime searches for the view in the following paths:
+
+  - Views/<controller_name>/Components/<view_component_name>/<view_name>
+  - Views/Shared/Components/<view_component_name>/<view_name>
+  
+The default view name for a view component is *Default*, which means your view file will typically be named *Default.cshtml*. You can specify a different view name when creating the view component result or when calling the ``View`` method.
+ 
+We recommend you name the view file *Default.cshtml* and use the *Views/Shared/Components/<view_component_name>/<view_name>* path. The ``PriorityList`` view component used in this sample uses *Views/Shared/Components/PriorityList/Default.cshtml* for the view component view.
+
+Invoking a view component
+-------------------------
+
+To use the view component, call ``@Component.InvokeAsync("Name of view component", <parameters>)`` from a view. The parameters will be passed to the ``InvokeAsync`` method.  The ``PriorityList`` view component developed in the article is invoked from the *Views/Todo/Index.cshtml* view file. In the following, the ``InvokeAsync`` method is called with two parameters:
```

Comment:
The name of a view component needs to be defined. See my comment with proposed text above.